### DISCUSSION TITLE
Copy / Pasting Snippets of Transcript

### DISCUSSION THREAD

**Discussion by jimparrillo** [lstrzepek/obsidian-yt-transcript#27] [Ideas]
I take a lot of lecture notes directly from YT and I would absolutely LOVE to be able to select text from the transcript and copy/paste into a note. At present, all you can do is copy all.

Thanks!!

**Comment by ichmoimeyo** [comment]
Based on https://github.com/lstrzepek/obsidian-yt-transcript/issues/18#issuecomment-1597620262 

.yt-transcript__transcript-block {
user-select: text;
}

I created ytt_snippet.css
![Screenshot 2023-08-23 185720](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/0dd5ea7d-e011-479a-95de-fadbe5fbacaa)
... selected some text with their timestamps
![Screenshot 2023-08-23 185226](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/f2424602-f91d-44a9-982b-283e4b136720)
... and dragged it  over into a new note
![Screenshot 2023-08-23 185331](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/d88c8be2-9f09-4ae8-82a1-a56ce5746a10)
... the timestamps **WORK** e.g. **00:13** starts playing @ beginning **00:13** in EXTERNAL YouTube
[Copy  Pasting Snippets of Transcript in YouTube EXTERNAL.md](https://github.com/lstrzepek/obsidian-yt-transcript/files/12428358/Copy.Pasting.Snippets.of.Transcript.in.YouTube.EXTERNAL.md)

\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

To play in **Obsidian** install Media Extended
https://github.com/aidenlx/media-extended
... because
https://github.com/aidenlx/media-extended/wiki/Media-Fragment-URI
... **BUT** the timestamps **DON'T WORK** e.g. **00:13** starts playing @ beginning **00:00** or cached YT.
![Screenshot 2023-08-23 191440](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/e59c2d63-1035-4d73-9172-2dc61af4ed10)
**NEED to find & replace ALL** 
**&t** 
**CHANGE TO** 
**#t**
![Screenshot 2023-08-23 191628](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/cf6b8b80-8653-41d1-b23b-b3e86f40706c)
... now the timestamps will work correctly :)
![Screenshot 2023-08-23 191747](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/0c9273cc-e449-4b60-92a1-5a8dec2aca0c)
[Copy  Pasting Snippets of Transcript in OBSIDIAN.md](https://github.com/lstrzepek/obsidian-yt-transcript/files/12428361/Copy.Pasting.Snippets.of.Transcript.in.OBSIDIAN.md)

**Reply by ichmoimeyo** [reply]
ALTERNATIVE 
 Install one of these chrome extensions from glasp.co

[YouTube Summary with ChatGPT](https://chrome.google.com/webstore/detail/youtube-summary-with-chat/nmmicjeknamkfloonkhhcjmomieiodli)

[Glasp: Social Web Highlight & YouTube Summary](https://chrome.google.com/webstore/detail/glasp-social-web-highligh/blillmbchncajnhkjfdnincfndboieik) [also has web highlighting & annotation]

**Copy and paste the transcript MANUALLY (includes timestamps)**
![Screenshot 2023-08-24 003307](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/2d65f5a5-7fa5-4ee0-8a4a-ba986844d9ba)
OR
select **markdown** in chrome extension's options
![Screenshot 2023-08-24 190459](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/4a0264f4-5857-49df-9867-72daf2dd0bc6)
and **CLICK ON -> Copy Transcript(Markdown) icon**
[PS: I'm also using **AI Summary** - Prompt: Summarize the following in an **ordered** list.]
![Screenshot 2023-08-24 190654](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/ab001d4e-46dd-474e-9536-12cd43bab2e2)
 into a new Obsidian note.
AND
Using Regex Find/Replace
https://github.com/Gru80/obsidian-regex-replace

FIND
/watch?v=undefined
REPLACE WITH
https://www.youtube.com/watch?v=kXeODjqvNAg
![Screenshot 2023-08-23 205427](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/32a7e5d0-d736-495f-b94a-870cc3a66404)

... the timestamps will play correctly externally in YouTube( if Media Enhanced plugin is disabled/not installed)
![Screenshot 2023-08-23 222027](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/c6a8f4d9-863e-4594-aced-f080609754bb)
[GLASP_transcript_PLAY_YT_EXTERNAL.md](https://github.com/lstrzepek/obsidian-yt-transcript/files/12427211/GLASP_transcript_PLAY_YT_EXTERNAL.md)

\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\

IF you want to play it in Obsidian ...
FIND
%t
REPLACE WITH
#t

AND
FIND
s)
REPLACE WITH
)

we now have  in edit mode ...
![Screenshot 2023-08-23 210234](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/7a079bd5-e24c-4b05-a365-cb6412c3a422)
WITH Media Extended installed
https://github.com/aidenlx/media-extended
... because
https://github.com/aidenlx/media-extended/wiki/Media-Fragment-URI
NOW in reading mode by clicking on **01.17** it starts playing at the correct time ...
![Screenshot 2023-08-23 210636](https://github.com/lstrzepek/obsidian-yt-transcript/assets/33708734/2d77a9ac-cd18-41a9-998f-1160de9101d7)
[GLASP_transcript_PLAY_OBSIDIAN.md](https://github.com/lstrzepek/obsidian-yt-transcript/files/12427230/GLASP_transcript_PLAY_OBSIDIAN.md)

**Reply by ichmoimeyo** [reply]
Another great way of watching a video & its transcript is to use the Hypothes.is based ...
https://docdrop.org/
e.g.
https://docdrop.org/video/kXeODjqvNAg/

PS: docdrop also has a great bookmarklet you can drag from its website to your bookmarks bar OR create yourself ...
Bookmark ANY page & then EDIT as follows:

Name: jhtd
URL:
javascript:const dropdocBaseURL="https://docdrop.org/video",url=new URL(location.href);switch(url.host){case"youtu.be":location.href=`${dropdocBaseURL}${url.pathname}`;break;case"m.youtube.com":case"www.youtube.com":location.href=`${dropdocBaseURL}/${url.searchParams.get("v")}`;break;default:alert("Not a recognized Youtube URL")}

**Comment by lstrzepek** [comment]
Now with version 0.9.1 it should be easy. Sorry to keep you waiting so long.
